nav2_common/nav2_common/launch/parse_multirobot_pose.py

```diff
@@ -12,66 +12,49 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-import sys
 from typing import Dict, Text
 
+import launch
+from launch.launch_context import LaunchContext
 import yaml
 
 
-class ParseMultiRobotPose:
-    """Parsing argument using sys module."""
+class ParseMultiRobotPose(launch.Substitution):
+    """
+    A custom substitution to parse the robots argument for multi-robot poses.
 
-    def __init__(self, target_argument: Text):
-        """
-        Parse arguments for multi-robot's pose.
+    Expects input in the format:
+    robots:="robot1={x: 1.0, y: 1.0, yaw: 0.0}; robot2={x: 2.0, y: 2.0, yaw: 1.5707}"
+    """
 
-        for example,
-        `ros2 launch nav2_bringup bringup_multirobot_launch.py
```

Comment:
Please add back in the instruction that shows example of XY/Yaw and XYZ/RPY so that folks can see how to set both (and that they _could_ set both). That was an intentional choice by the authors